elasticapm/instrumentation/packages/urllib3.py

```diff
@@ -61,7 +61,12 @@ def update_headers(args, kwargs, instance, transaction, trace_parent):
     :param trace_parent: the TraceParent object
     :return: an (args, kwargs) tuple
     """
-    if len(args) >= 4 and args[3]:
+    from urllib3._version import __version__ as urllib3_version
+
+    if urllib3_version.startswith("2") and len(args) >= 5 and args[4]:
+        headers = args[4].copy()
+        args = tuple(itertools.chain((args[:4]), (headers,), args[5:]))
+    elif len(args) >= 4 and args[3]:
```

Comment:
I was initially surprised to see version dependent code here since urllib3 2.0 is supposed to be mostly compatible with urllib3 1.26. Looking more closely, I understand now that you need to set trace headers irrelevant of the way they were set by the caller (arg, kwarg, instance parameter). Since urllib3 2.0 added a `body` field, it moved `headers` from 4th positional parameter to 5th positional parameter, and you have to detect this. cc @sethmlarson 

Is my understanding right? Should you update the docstring to mention urllib3 2.0?

Reply:
Yes, you understood correctly. I probably should throw a comment in there for better documentation.